### PR TITLE
Import scan oxo

### PR DESCRIPTION
**Pull request by elyousfi5**
[Screencast from 2024-05-22 10-15-39.webm](https://github.com/Ostorlab/oxo/assets/144013278/c2452938-7f6d-46e7-a1bf-f4962c0837dd)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -61,6 +61,9 @@ install_requires =
     alembic
     py-ubjson
     nats-py
+    graphene-file-upload
+    Flask-GraphQL
```
Comment (PiranhaSa) [inline setup.cfg:65]:
```suggestion
    flask-graphql
```

File: src/ostorlab/cli/__init__.py [hunk 2]
```diff
@@ -7,5 +7,15 @@
...
-__all__ = ["scan", "auth", "agent", "vulnz", "agentgroup", "ci_scan", "scanner"]
+__all__ = [
+    "scan",
+    "auth",
+    "agent",
+    "vulnz",
+    "agentgroup",
+    "ci_scan",
+    "scanner",
+    "server",
```
Comment (3asm) [inline src/ostorlab/cli/__init__.py:20]:
```suggestion
    "serve",
```

File: src/ostorlab/cli/server/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+"""Module for the root command: scan."""
```
Comment (3asm) [inline src/ostorlab/cli/server/__init__.py:1]: Replace server by serve

File: setup.cfg [hunk 4]
```diff
@@ -61,6 +61,10 @@ install_requires =
     alembic
     py-ubjson
     nats-py
+    graphene-file-upload
+    flask-graphql
+    graphene-sqlalchemy
+    requests==2.26.0
```
Comment (3asm) [inline setup.cfg:67]: Wrap behind a different target.

File: src/ostorlab/cli/server/run/run.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+"""Module for the ostorlab server run command."""
+
+import click
+
+from ostorlab.cli.server.server import server
+from ostorlab.runtimes.local.app import app
+
+
+@server.group(invoke_without_command=True)
+@click.option("--host", default="127.0.0.1", help="The host to run the Flask app on.")
```
Comment (3asm) [inline src/ostorlab/cli/server/run/run.py:10]:
```suggestion
@click.option("--host", default="0.0.0.0", help="The host to run the Flask app on.")
```

File: src/ostorlab/cli/server/run/run.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+@click.option("--port", default=5000, help="The port to run the Flask app on.")
```
Comment (3asm) [inline src/ostorlab/cli/server/run/run.py:11]:
```suggestion
@click.option("--port", default=3420, help="The port to run the Flask app on.")
```

File: src/ostorlab/cli/server/run/run.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+"""Module for the ostorlab server run command."""
+
+import click
+
+from ostorlab.cli.server.server import server
```
Comment (3asm) [inline src/ostorlab/cli/server/run/run.py:5]: We don't need run. We need `serve`.

File: src/ostorlab/runtimes/local/app/app.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import flask
+from graphene_file_upload import flask as graphene_upload_flask
+
+from ostorlab.runtimes.local.app.schema import schema
+
+flask_app = flask.Flask(__name__)
+
+flask_app.add_url_rule(
+    "/oxo_graphql",
```
Comment (3asm) [inline src/ostorlab/runtimes/local/app/app.py:9]:
```suggestion
    "/graphql/",
```

File: src/ostorlab/runtimes/local/app/utils.py [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+from ostorlab.runtimes.local.models import models
+from ostorlab.utils import risk_rating
+
+SCAN_JSON = "scan.json"
+ASSET_JSON = "asset.json"
+VULNERABILITY_JSON = "vulnerability.json"
+
+
+def import_scan_from_bytes(
```
Comment (3asm) [inline src/ostorlab/runtimes/local/app/utils.py:17]:
```suggestion
def import_scan(
```

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/runtimes/local/app/utils.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    session: models.Database,
+    file_data: bytes,
+    append_to_scan: Optional[models.Scan] = None,
+) -> None:
```
Comment (oussamaessaji) [inline src/ostorlab/runtimes/local/app/utils.py:21]: docstring

File: src/ostorlab/runtimes/local/app/app.py (hunk 8 above)
Comment (ErebusZ) [reply src/ostorlab/runtimes/local/app/app.py:9]: maybe this should be oxo for consistancy?

File: src/ostorlab/runtimes/local/app/utils.py [hunk 11]
```diff
@@ -0,0 +1,79 @@
+"""utils module: contains utility functions to export/import the scan details."""
```
Comment (3asm) [inline src/ostorlab/runtimes/local/app/utils.py:1]: Change file name to import_utils.

File: tests/runtimes/local/app/utils_test.py [hunk 12]
```diff
@@ -0,0 +1,29 @@
+"""Unit tests for the import_scan module."""
+
+from ostorlab.runtimes.local.app import utils
+from ostorlab.runtimes.local.models import models
+from ostorlab.utils import risk_rating
+
+
+def testImportScan_always_shouldImportScan(
```
Comment (3asm) [inline tests/runtimes/local/app/utils_test.py:8]: I don't see any unit test related to the graphql endpoint.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: tests/runtimes/local/app/utils_test.py (hunk 12 above)
Comment (elyousfi5) [reply tests/runtimes/local/app/utils_test.py]: added, please check again

**Review by 3asm — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**